pkg/secrets/config/config.go

```diff
@@ -49,10 +54,22 @@ func NewFromEnv() (*Config, error) {
 		return nil, err
 	}
 
+	timeout := DEFAULT_TIMEOUT
+	if envTimeout, _ := strconv.Atoi(os.Getenv("TIMEOUT")); envTimeout != DEFAULT_TIMEOUT && envTimeout != 0 {
+		timeout = envTimeout
+	}
+
+	retryCountLimit := DEFAULT_RETRY_COUNT_LIMIT
+	if envRetryCountLimit, _ := strconv.Atoi(os.Getenv("RETRY_COUNT_LIMIT")); envRetryCountLimit != DEFAULT_RETRY_COUNT_LIMIT && envRetryCountLimit != 0 {
```

Comment:
why do you check for value != 0?
Is this some kind of input validation?
If so, it should be clear. Since retry = 0 might be a valid input.
If it is to avoid the default value in case of parsing error, better check for `err != nil`, then use the default.

Reply:
if the user didn't provide in these env vars value they would programmatically equate to 0 so this is to ensure that this doesn't happen